src/merkle/mmr/full.rs

```diff
@@ -246,6 +255,108 @@ where
     }
 }
 
+// ITERATOR
+// ===============================================================================================
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Node {
+    Parent {
+        value: Word,
+        left: Word,
+        right: Word,
+    },
+    Leaf {
+        value: Word,
+    },
+}
```

Comment:
Similar comment to https://github.com/0xPolygonMiden/crypto/pull/117#discussion_r1154804910.

Reply:
rebased this PR on #117 , now it uses `InnerNodeInfo`.